# STAT 207 Group Lab Assignment 1 - [10 total points]

## Exploring your own dataset!

<hr>

## <u>Lab Grading</u>:

Should we grade your submission?  If not, write the netID of the submission to be graded.  (Note: We will only grade one assignment per group, and we'll pick the first one that says we should grade that submission.)

*For example*, you might respond: **grade this submission** or **my submission is under netID jdeeke**

dgeni2

## <u>Purpose</u>:
You should work in groups of 2-3 on this report. The purpose of this group lab assignment is to download your own dataset that you find interesting and perform some preliminary descriptive analytics on this dataset. Also, you can take this opportunity to get to know some of your classmates! 
<hr>

## <u>Assignment Instructions</u>:

### [1 point] Contribution Report
These contribution reports should be included in all group lab assignments. In this contribution report below you should list of the following:
1. The netID for the lab submission to be graded.  (Some groups have each member create their own version of the document, but only one needs to be submitted for grading.  Other groups have only one member compose and submit the lab.)
2. Names and netIDs of each team member.
3. Contributions of each team member to report.

*For example:*

*<u>Teammates:</u>*

*doe105 should be graded.  John Smith (smith92) & Jane Doe (doe105) worked together on all parts of this lab assignment*, 

OR

*doe105 should be graded*

<i>1. John Smith (smith92) was the navigator</i>

<i>2. Jane Doe (doe105) was the driver</i>

dgeni2 should be graded
1. Colin Richardson (colinhr3) was the navigator
2. Dulf Vincent Genis (dgeni2) was the driver

### Group Roles

You are expected to work in groups of 2-3 on this report.  Since you are working in groups, you may find it helpful to have specified roles.  These roles will likely be helpful in later labs.  Below, I provide roles that can be used for groups of 2 and for groups of 3.  I encourage you to switch roles within this lab report, as possible.  I also encourage you to switch roles for each subsequent lab, as possible, based on your group membership.  

#### Groups of 2

* **Driver**: This student will type the report.  While typing the report, you may be the one who is selecting the functions to apply to the data.
* **Navigator**: This student will guide the process of answering the question.  Specific ways to help may include: outlining the general steps needed to solve a question (providing the overview), locating examples within the course notes, and reviewing each line of code as it is typed.

#### Groups of 3

* **Driver**: This student will type the report.  They may also be the one to select the functions to apply to the data.
* **Navigator**: This student will guide the process of answering the question.  They may select the general approach to answering the question and/or a few steps to be completed along the way. 
* **Communicator**: This student will review the report (as it is typed) to ensure that it is clear and concise.  This student may also locate relevant examples within the course notes that may help complete the assignment.

<hr>

### Imports

In [4]:
#Run this
import pandas as pd                    # imports pandas and calls the imported version 'pd'
import matplotlib.pyplot as plt        # imports the package and calls it 'plt'
import seaborn as sns                  # imports the seaborn package with the imported name 'sns'
sns.set()  

<hr>

## Initial Dataset Analysis

### 1. [3 points] Load an Interesting Dataset

1. Go to https://www.kaggle.com/datasets?fileType=csv&sizeStart=0%2CKB&sizeEnd=100%2CKB. Browse through the available datasets and find one that you think is interesting *that has at least one categorical variable* and *at least one quantitative variable*.
2. Once you have selected a dataset you'd like to explore, select "Download" to download the csv file.
3. On your desktop, move this csv into the same folder that this Jupyter notebook is located in.
4. Finally, read this csv file into a pandas dataframe and display the first five rows.

In [8]:
paris2024csv = pd.read_csv("olympics2024.csv")
paris2024csv.head(5)

,Rank,Country,Country Code,Gold,Silver,Bronze,Total
0,1,United States,US,40,44,42,126
1,2,China,CHN,40,27,24,91
2,3,Japan,JPN,20,12,13,45
3,4,Australia,AUS,18,19,16,53
4,5,France,FRA,16,26,22,64


### 2. [3 points] Dataset Description

**a)** Give the link for the dataset that you downloaded.

https://www.kaggle.com/datasets/berkayalan/paris-2024-olympics-medals?resource=download

**b)** How many rows and how many columns are in the dataset?

In [6]:
paris2024csv.shape

(91, 7)

91 rows
7 columns

**c)** What is the unit of observation for this dataset?  In other words, each row contains variables recorded for an observation.  How would you define a general observation?

The different countries that participated in the olympics and the number of medals each one accumulated.

### 3. [3 points] Descriptive Summaries

**a)** Define a question.

Considering the available data, ask a question that can be addressed with the data.  

What is the most number of medals won this year?

**b)** Answer your question.

Now, generate a descriptive summary that addresses at least some part of your question from 3a.  You don't need to completely answer or address your question if you pose a complicated question, but your analysis should relate to the original question.

Your descriptive summary should include at least one of a numerical summary measure or a visualization (graph).

In [20]:
paris2024csv["Total"].max()

126

The most medals won this year by a country was 126.